app/controllers/spree/adyen_redirect_controller.rb

```diff
@@ -7,10 +7,10 @@ class AdyenRedirectController < StoreController
 
     # This is the entry point after an Adyen HPP payment is completed
     def confirm
-      ActiveRecord::Base.transaction do
+      @order.with_lock do
         # reload order as it might have changed since previously loading it
```

Comment:
This comment could be moved up and it should be noted that there is an implicit reload in the lock